addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/PresenceDetection.java

```diff
@@ -518,32 +563,34 @@ public void stopAutomaticRefresh() {
             future.cancel(true);
             refreshJob = null;
         }
-        enableDHCPListen(false);
+        if (cachedDestination != null) {
+            disableDHCPListen(cachedDestination);
+        }
     }
 
     /**
-     * Enables/Disables listing for dhcp packets to figure out if devices have entered the network. This does not work
+     * Enableslisting for dhcp packets to figure out if devices have entered the network. This does not work
      * for iOS devices. The hostname of this network service object will be registered to the dhcp request packet
      * listener if enabled and unregistered otherwise.
      *
-     * @param enabled Enable/Disable the dhcp listen service for this hostname.
+     * @param destinationAddress the InetAddress to listen for.
      */
-    private void enableDHCPListen(boolean enabled) {
-        if (enabled) {
-            try {
-                if (DHCPListenService.register(destination.getHostAddress(), this).isUseUnprevilegedPort()) {
-                    dhcpState = "No access right for port 67. Bound to port 6767 instead. Port forwarding necessary!";
-                } else {
-                    dhcpState = "Running normally";
-                }
-            } catch (SocketException e) {
-                logger.warn("Cannot use DHCP sniffing.", e);
-                useDHCPsniffing = false;
-                dhcpState = "Cannot use DHCP sniffing: " + e.getLocalizedMessage();
+    private void enableDHCPListen(InetAddress destinationAddress) {
+        try {
```

Comment:
Could you please call `disableDHCPListen` as first method? `DHCPListenService.unregister` should be called on a potential old address before listing to this new address.